torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -73,7 +73,8 @@ std::map<at::ScalarType, ncclDataType_t> ncclDataType = {
 // Helper function that gets the data type and issues error if not supported
 ncclDataType_t getNcclDataType(at::ScalarType type) {
   auto it = ncclDataType.find(type);
-  TORCH_CHECK(
+  TORCH_CHECK_WITH(
```

Comment:
Well, maybe use `C10_THROW_ERROR` as well?

Reply:
Need to use `TORCH_CHECK_WITH` here since the exception is thrown conditionally on `it != ncclDataType.end()`. `C10_THROW_ERROR` throws unconditionally.

Reply:
I see.